### PR TITLE
fix: dot-net cli now builds tree instead of flat list

### PR DESCRIPTION
**Pull request by eavichay**
This change enables inspection of projects created with dotnet-cli (most used in linux/osx dev machines). It reads the project.asset.json file and extracts the whole dependency tree from the manifest file.

### CODE REVIEW

File: lib/formats/dotnet-cli-parser.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  // restructure tree with "from" chains.
+  function buildPath(node, from) {
+    node.from = node.from.concat(from);
+    Object.keys(node.dependencies).forEach(function (key) {
+      var depFrom = node.from.concat(node.name + '@' + node.version);
+      buildPath(node.dependencies[key], depFrom);
+    });
+  }
+
+  var pathedTree = JSON.parse(JSON.stringify(packageTree.dependencies));
```
[the remark this answers is not in the document]
Comment (eavichay) [reply lib/formats/dotnet-cli-parser.js]: The tree is built by objects and references. In order to disconnect the references and do an object deep clone, this is the simplest way to do it.

File: lib/formats/dotnet-cli-parser.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+var Dependecy = require('../dependency');
+var debug = require('debug')('snyk');
+
+function convertFromPathSyntax(path) {
+  var name = path.split('/').join('@'); // posix
+  name = name.split('\\').join('@'); // windows
+  return name;
+}
```
[the remark this answers is not in the document]
Comment (eavichay) [reply lib/formats/dotnet-cli-parser.js]: I read that nuget support posix and windows paths - projects can be developed in two different environments. I may want to scan an imported project from windows to linux and vice versa).

File: lib/formats/dotnet-cli-parser.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrukh) [reply lib/formats/dotnet-cli-parser.js]: Then please add a comment explaining this :) Easy to misunderstand

File: lib/index.js [hunk 3]
```diff
@@ -5,14 +5,18 @@ var Dependecy = require('./dependency');
...
+var projectJsonFormatParser = require('./formats/dotnet-cli-parser');
 
 function determineManifestType (filename) {
   switch (true) {
-    case /.json$/.test(filename): {
+    case /project.json$/.test(filename): {
       return 'project.json';
     }
+    case /project.assets.json$/.test(filename): {
+      return 'dotnet-cli';
```
[the remark this answers is not in the document]
Comment (eavichay) [reply lib/index.js]: the cli filters the file names and initiates nuget plugin. some of the plugin's test are based on filenames like _1_project.json and others and thus passing the regexp detection.
In a real-life scenario the filter is one level above, in snyk/detect.

File: lib/formats/dotnet-cli-parser.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrukh) [reply lib/formats/dotnet-cli-parser.js]: Got it!

**Pull request merged**